src/main/java/com/sourcegraph/webhook/WebhookListener.java

```diff
@@ -35,8 +38,30 @@
     }
 
     @EventListener
-    public void onPullRequestEvent(PullRequestActivityEvent event) {
-        String key = "pr:activity:status";
+    public void onPullRequestEvent(PullRequestEvent event) {
+        String key = "pr:event";
+
+        // When review status changes, two events are fired in most cases
+        // 1. A PR Activity Event
+        // 2. A PR Participant Event
+        // We only want to forward the Activity events.
+        // However, when moving from Needs Works to Unapproved, no activity event is fired
+        // and so in this one case we want to forward the Participant Event
+        if (event instanceof PullRequestParticipantStatusUpdatedEvent) {
+            PullRequestAction action = event.getAction();
+            PullRequestParticipantStatus previousStatus = ((PullRequestParticipantStatusUpdatedEvent) event).getPreviousStatus();
+            if (action != PullRequestAction.UNAPPROVED) {
+                return;
+            }
+            if (previousStatus != PullRequestParticipantStatus.NEEDS_WORK) {
+                return;
+            }
+            key =  "pr:participant:status";
+        }
+
+        if (event instanceof PullRequestActivityEvent) {
+            key = "pr:activity:event";
```

Comment:
The old default key was `pr:activity:status` — is this change intentional?

Reply:
Yes, that's just cleaning things up. Our current go code ignores it anyway but an incoming PR uses the new values.